Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -31,76 +38,133 @@ namespace ROS2::Utils
         return r;
     }
 
+    AZStd::string GetProductAsset(const AZ::Data::AssetId& assetId , const AZ::TypeId typeId)
+    {
+        AZStd::vector<AZ::Data::AssetInfo> productsAssetInfo;
+        using AssetSysReqBus = AzToolsFramework::AssetSystemRequestBus;
+        bool ok{ false };
+        AssetSysReqBus::BroadcastResult(
+            ok, &AssetSysReqBus::Events::GetAssetsProducedBySourceUUID, assetId.m_guid, productsAssetInfo);
+        if (ok)
+        {
+            for (auto& product : productsAssetInfo)
+            {
+                if (product.m_assetType == typeId)
+                {
+                    AZStd::string assetPath;
+                    AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                        assetPath, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetPathById, product.m_assetId);
+                    return assetPath;
```

Comment:
I can see that multiple assets are created (including azmeshes) for larger files.  
I thought the first one was guaranteed to be the main one (I've tested this code on such cases).
Could you provide me details on "prefab" generation?

Reply:
I'll try to give a brief detail on prefab generation.  Its a similar concept to URDF import.  In URDF, there is a URDF file that describes the transform heirarchy and the relation of the pieces to each other, and by convention, it references the meshes as individual DAE or STL files.   So the URDF itself is the "scene description" file.

In many other pipelines, however (common in games and other industry) the entire scene is assembled in the DCC tool like blender or maya, and the rig lives entirely in one file (minus optional attachments).  So for example imagine a blender or maya scene where the entire robot, all the joints, all the settings, and all the parts are inside the one DAE or FBX file (as seperate objects in a heirarchy of transforms).

the Prefab auto generation rule is on by default for all meshes (you'll see it show up as a prefab group rule in the assetinfo files or in the prefab tab in the GUI).  When active what it does is
* Create mesh export rules for each individual object in the scene heirarchy inside the FBX/Dae/STL/etc file, so that each seperate object from the DCC tool gets seperated into its own mesh file.  These meshes are always exported in mesh local space - that is, always with their pivot at the origin 0,0,0, with their transform in the heirarchy inversed out of their vertex data.
* Exports a .procprefab file as part of the export.  A procprefab file is an identical file format as normal prefab files, that is, a heirarchy of transforms of entities with components.  It assigns the transforms found in the DCC file, similar to how the URDF converter assigns the transforms found in the URDF file.  It also assigns the meshes it extracted from the DCC file into mesh components.  Again, very similar to robot importer.  The result is a .procprefab file which the user can then drag into the scene to spawn the scene heirarchy in the editor.    They can also create a prefab file that references the procprefab file and add overrides.

Both systems (Procedural prefab generation and URDF import) serve similar functions - make a prefab based on whats in the scene file.  however, in general, in URDF land, the URDF file itself is the scene file and the meshes it needs are already split into separate mesh files.  Wheras in procprefab land, the single FBX contains the entire scene (there is no URDF) and it needs to split them.

So in this case, I would recommend that if you are making a tool which imports a URDF, you can assume that any source DAE/FBX/STL referenced by it are already single mesh pieces and do not need a procprefab generaetd for each of them, in which case its probably best to create a scenesettings file that turns off the procprefab rule for them, so that they only generate the mesh files you expect and don't generate .procprefab files in addition.

An example here to illustrate.
Imagine you have a scene that has two cubes standing on top of each other in one dcc file, lets call it "shape_tower.fbx"
```
base_transform <0,0,0> <-- just a parent transform that is empty
     cube1 <--unit cube mesh, <0, 0.5, 0> so its sitting on top of origin 
        sphere1 <-- unit sphere mesh <0, 0.5, 0> relative to parent
```
If you allow the default processing to happen without changing anything, this 'scene' will be exported as
```
   shape_tower.atommodel <-- contains all the vertices from cube1, sphere1, all welded together in their world space poses, basically the entire scene as 1 mesh
   cube1_procprefab_(WART).atommodel <-- the cube only, centered around the origin, with no offset
   sphere1_procprefab_(WART).atommodel <-- the sphere only, centered around the origin, with no offset.
   shape_tower.procprefab <-- a prefab file, contianing 3 transforms and 2 mesh components with offsets set, meshes assigned to cube1_procprefab, sphere1.procprefab
```

If you delete the prefab group in the prefab tab, it will delte the procprefab mesh rules and you'll end up with it only making shape_tower.atommodel,

Reply:
Totally agree, I did not know that such workflow exists, and it is default behavior.
Please find a new commit in PR, that disables procedural assets at all during URDF import.
It generate no procedural prefab (it also stops to create visual meshes if there is collider only mesh):
![image](https://user-images.githubusercontent.com/3209244/228573534-0115883c-7140-461b-a460-9bdd64406fec.png)
I still think it create a lot of product assets in local coordinate system (as you explained):
![image](https://user-images.githubusercontent.com/3209244/228573857-ce18bab4-f922-449c-83d6-4aa1f5e515d4.png)

Reply:
I'm surprised it does that still, actually, it would indicate that somehow the proc prefab rule is still making mesh groups for the finger even tho its been turned off.
all of the "default_xxxxxxx_(some guid)" files are procprefab stuff as far as I know